server/controllers/UserControllers.js

```diff
@@ -0,0 +1,361 @@
+      return res.status(400).send({
+        message: 'limit and offset must be an integer'
+      });
+    }
+    return User.findAndCount({
+      limit,
+      offset,
+      attributes: ['id', 'fullname', 'username', 'email', 'RoleId'],
+    })
+    .then(user => res.status(200).send({
+      pagination: {
+        row: user.rows,
+        paginationDetails: pagination(user.count, limit, offset)
+      }
+    }))
+    .catch(error => res.status(403).send(error.message));
+  },
+
+
+  /**
+   * @param {Object} req
+   * @param {Object} res
+   * @returns {Object} returns a response object containing the list of user registered
+   */
+  searchUsers(req, res) {
+    let searchKey = '%%';
+    if (req.query.q) {
+      searchKey = `%${req.query.q}%`;
+    }
+
+    return User
+    .findAll({
+      where: { username: {
+        $iLike: searchKey
+      } },
+      order: [['createdAt', 'DESC']]
+    })
+    .then(users => res.status(200).send({
+      status: 'OK',
+      count: users.length,
+      userList: users.map(user => (
+        {
+          id: user.id,
+          fullname: user.fullname,
+          username: user.username,
+          email: user.email,
+          role: user.RoleId,
+          created_at: user.createdAt
+        }))
+    }))
+    .catch(error => res.status(400).send(error.message));
+  },
+
+  /**
+   * @param {Object} req
+   * @param {Object} res
+   * @returns {Object} returns a response object
+   */
+  getUser(req, res) {
+    if (isNaN(req.params.id)) {
+      return res.status(400).send({
+        status: 'error',
+        message: 'id must be a number'
+      });
+    }
+    return User
+    .findById(req.params.id)
+    .then((user) => {
+      if (!user) {
+        return res.status(404).send({
+          status: 'error',
+          message: 'User not found',
+        });
+      }
+      req.user = user;
+      return res.status(200).send({
+        id: user.id,
+        fullname: user.fullname,
+        username: user.username,
+        email: user.email,
+        role: user.RoleId,
+        createdAt: user.createdAt,
+        updatedAt: user.updatedAt
+      });
+    })
+    .catch(() => res.status(400).send({
+      message: 'Invalid param'
+    }));
+  },
+
+  /**
+   * @param {Object} req
+   * @param {Object} res
+   * @returns {Object} returns a response object
+   */
+  updateUser(req, res) {
+    if (isNaN(req.params.id)) {
+      return res.status(400).send({
+        message: 'Id must be a number'
+      });
+    }
+    return User
+    .findById(req.params.id)
+    .then((user) => {
+      if (Number(req.params.id) !== user.id) {
+        return res.status(403).send({
+          message: 'Access denied',
+        });
+      }
+      User.findAll({
+        where: {
+          $or: [{
+            email: req.body.email
+          }, {
+            username: req.body.username
+          }]
+        }
+      })
+        .then((AuthenticatedUser) => {
+          if (AuthenticatedUser.length !== 0
```

Comment:
You can just simply call `AuthenticatedUser.length` instead of the explicit check for if it's not zero